go/oasis-node/cmd/config/migrate/migrate.go

```diff
@@ -0,0 +1,617 @@
+
+		if file, ok := m(log)["file"]; ok {
+			mLog["file"] = file
+			delete(m(log), "file")
+		}
+
+		if format, ok := m(log)["format"]; ok {
+			mLog["format"] = format
+			delete(m(log), "format")
+		}
+
+		if level, ok := m(log)["level"]; ok {
+			if mLevel, isMap := level.(map[string]interface{}); isMap {
+				// Copy the map.
+				mkSubMap(mLog, "level")
+				for k, v := range mLevel {
+					m(mLog["level"])[k] = v
+				}
+			} else {
+				if sLevel, isString := level.(string); isString {
+					// If only a single level is given instead of the map,
+					// convert it into a map with only the default level.
+					mkSubMap(mLog, "level")
+					m(mLog["level"])["default"] = sLevel
+				} else {
+					logger.Warn("log.level is malformed, ignoring")
+				}
+			}
+			delete(m(log), "level")
+		}
+	}
+
+	debug, ok := oldCfg["debug"]
+	if ok {
+		logger.Info("debug.* is now common.debug.*")
+		mkSubMap(newCfg, "common")
+		mkSubMap(m(newCfg["common"]), "debug")
+		mDebug := m(m(newCfg["common"])["debug"])
+
+		if rlimit, ok := m(debug)["rlimit"]; ok {
+			mDebug["rlimit"] = rlimit
+			delete(m(debug), "rlimit")
+		}
+
+		if allow_root, ok := m(debug)["allow_root"]; ok {
+			mDebug["allow_root"] = allow_root
+			delete(m(debug), "allow_root")
+		}
+	}
+
+	pprof, ok := oldCfg["pprof"]
+	if ok {
+		if bind, ok := m(pprof)["bind"]; ok {
+			logger.Info("pprof.bind is now pprof.bind_address")
+			mkSubMap(newCfg, "pprof")
+			m(newCfg["pprof"])["bind_address"] = bind
+			delete(m(pprof), "bind")
+		}
+	}
+
+	ias, ok := oldCfg["ias"]
+	if ok {
+		mkSubMap(newCfg, "ias")
+		mIAS := m(newCfg["ias"])
+
+		if proxy, ok := m(ias)["proxy"]; ok {
+			if address, ok := m(proxy)["address"]; ok {
+				logger.Info("ias.proxy.address is now ias.proxy_address")
+				mIAS["proxy_address"] = address
+				delete(m(proxy), "address")
+			}
+		}
+
+		if debug, ok := m(ias)["debug"]; ok {
+			if skip_verify, ok := m(debug)["skip_verify"]; ok {
+				logger.Info("ias.debug.skip_verify is now ias.debug_skip_verify")
+				mIAS["debug_skip_verify"] = skip_verify
+				delete(m(debug), "skip_verify")
+			}
+		}
+	}
+
+	genesis, ok := oldCfg["genesis"]
+	if ok {
+		mkSubMap(newCfg, "genesis")
+		mGen := m(newCfg["genesis"])
+
+		if file, ok := m(genesis)["file"]; ok {
+			mGen["file"] = file
+			delete(m(genesis), "file")
+		}
+	}
+
+	// If we deleted any keys, make sure we don't retain empty structures.
+	pruneEmptyMaps(oldCfg)
+	// Also prune new config.
+	pruneEmptyMaps(newCfg)
+
+	// Check for options that are only available on the command-line.
+	if _, ok = oldCfg["debug"]; ok {
+		logger.Warn("note that some debug.* options are from now on only available on the command-line")
+	}
+	if _, ok = oldCfg["grpc"]; ok {
+		logger.Warn("note that grpc.* options are from now on only available on the command-line")
+	}
+	if _, ok = oldCfg["metrics"]; ok {
+		logger.Warn("note that metrics.* options are from now on only available on the command-line")
```

Comment:
Oh, didn't notice that before, why is this the case, doesn't it make sense to have them in the config file?

Reply:
I guess we could move them into the config in a separate PR...